src/data/array_interface.h

```diff
@@ -40,7 +41,7 @@ struct ArrayInterfaceErrors {
     return str.c_str();
   }
   static char const* Version() {
-    return "Only version 1 of `__cuda_array_interface__' is supported.";
+    return "Only version 1 and 2 of `__cuda_array_interface__' are supported.";
```

Comment:
Does version 1 still work with the new code?

Reply:
Yes, it doesn't have breaking change.

Reply:
Do we have a test for version 1?

Reply:
Not yet.  I got confused by the various document.  Version 2 should contain just clarification change.  So far we don't check for the version number due to inconsistent between `__array_interface__` and `__cuda_array_interface__`.  So this check is unused.

Also there's version 3 according to numba.

On latest pydata numba doc, it's still version 2, but on stable numba doc, it's already on version 3.

https://numba.readthedocs.io/en/stable/cuda/cuda_array_interface.html
http://numba.pydata.org/numba-doc/latest/cuda/cuda_array_interface.html

Reply:
There are a few things I need to do:

- Find a way to distinguish `__array_interface__` and `__cuda_array_interface__`.  This can be done by using a specific `adapter`.
- Go through version 3 and support it.  This will require us actually check the version since it adds a new field called `stream`.

Reply:
Thanks for providing the context.